### PR TITLE
header file added

### PR DESCRIPTION
**Pull request by AbhinandSuresh**

### CODE REVIEW

File: msgType.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	joinAccept,			//used in pairing mode for join accept
+	joinConfirm,		//used in pairing mode for join confirmation
+	configReq,			//used in configuration mode for requesting configuration message
+	configMsg,			//used in configuration mode for sending configuration message
+	configAck,			//used in configuration mode for acknowledging configuration successfull
+	sensorData,			//used in monitoring mode for sending sensor data and battery level
+	ack,				//used for acknowledging a message received
+	motorStatus,		//used for acknowleging status of motor
+	alarm,				//used to send warning messages/error messages
+	initController		//used to notify level monitor that the controller is recently initiated
```
Comment (greatinbaby) [inline msgType.h:14]: Better to use initNotify

File: msgType.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+}msgType; 
+
+typedef enum devType {
+	monitor = 0x01,
+	controller,
+	relayUnit
+}devType;
+
+typedef enum response {
```
Comment (greatinbaby) [inline msgType.h:23]: Is it showing result/error, then better use errorno
Status as 0 and others as numbers

File: msgType.h [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	failure = 0x00,
+	success
+}response;
+
+typedef enum motorState {
```
Comment (greatinbaby) [inline msgType.h:28]: Use deviceOn etc for avoiding confusion

File: msgType.h [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+typedef struct sensorData {
+	float level;
+	float batteryInfo;
+}sensorData;
+
+/*
+* Message type: acknowledgment
+* response : Acknowledgment response
+*/
+typedef struct ack {
```
Comment (greatinbaby) [inline msgType.h:103]: Did not require

File: msgType.h [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+typedef struct joinAccept {
+	unsigned char devID[4];
+	unsigned char secretKey[16];
+	}joinAccept;
+
+/*
+* Message type: Join Confirm
+* joinAck: Acknowledgement for joining confirmation
+*/
+typedef struct joinConfirm {
```
Comment (greatinbaby) [inline msgType.h:55]: ClientResponse?? One response message is enough for all

File: msgType.h [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+*/
+typedef struct ack {
+	unsigned char response;
+}ack;
+
+/*
+* Message type: Motor /Pump Status
+* status: Working status of the motor
+*/
+typedef struct motorStatus {
```
Comment (greatinbaby) [inline msgType.h:111]: Why need astruct here

File: msgType.h [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+	unsigned char motorState;
+}motorStatus;
+
+/*
+* Message type: alarm
+* dryRun: Warning to alarm a dry run state
+* lowBattery: Warning to alarm a lowBattery state
+* sensorErr: Warning to alarm a sensor error state
+*/
+typedef struct alarm {
```
Comment (greatinbaby) [inline msgType.h:121]: Not needed

File: msgType.h [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+	unsigned char dryRun;
+	unsigned char lowBattery;
+	unsigned char sensorErr;
+}alarm;
+
+/*
+* Message type: Controller initialization / Restart 
+* controllerRestart: Flag of controller restarted recently
+*/
+typedef struct initController {
```
Comment (greatinbaby) [inline msgType.h:131]: Not need

**Review by greatinbaby — CHANGES_REQUESTED** [review]

File: msgType.h (hunk 1 above)
Comment (AbhinandSuresh) [reply msgType.h:14]: ok

File: msgType.h (hunk 3 above)
Comment (AbhinandSuresh) [reply msgType.h:28]: ok

File: msgType.h (hunk 2 above)
Comment (AbhinandSuresh) [reply msgType.h:23]: it is responses for acknowledging join requests, acknowledgements of messages etc.
Error states are dealed here. shall make the necessary changes

Comment (AbhinandSuresh) [reply msgType.h:23]: it is responses for acknowledging join requests, acknowledgements of messages etc.
Error states are dealt with at this position. shall make the necessary changes.

File: msgType.h (hunk 3 above)
Comment (AbhinandSuresh) [reply msgType.h:28]: typedef enum motorState {
	DEVICE_ON = 0x00,
	DEVICE_OFF,
	DRY_RUN
}motorState;

File: msgType.h (hunk 4 above)
Comment (AbhinandSuresh) [reply msgType.h:103]: this is to send acknowledgments when a message is received., isn't it really irrelevant? please re-review.

File: msgType.h (hunk 5 above)
Comment (AbhinandSuresh) [reply msgType.h:55]: ok

File: msgType.h (hunk 6 above)
Comment (AbhinandSuresh) [reply msgType.h:111]: duplicate found. made necessary changes.

File: msgType.h (hunk 7 above)
Comment (AbhinandSuresh) [reply msgType.h:121]: ok

**Review by greatinbaby — APPROVED** [review]

**Review by sujithnair1 — APPROVED** [review]

**Pull request merged**